fetch-git/README.md

```diff
@@ -112,6 +112,11 @@ to behave responsibly, is also immutable and pure. Note: this
 can probably only be feasibly true in an organization with strict
 repository protection!
 
+This purity allows for reliable caching. The upstream fetchGit, by
+contrast, allows fetching the tip of any branch (or arbitrary ref) and
+has a convenience cache with a configurable TTL unless a specific
+commit is specified.
```

Comment:
From this description, it sounds like `builtins.fetchGit` also will give you reliable caching as long as a specific commit is specified. Is that correct? And that the only edge that `fetch-git` has (in this area) is that it can give reliable caching also by specifying a git tag?

If my understanding is correct, I think this could be made even more clear. So for example if you mostly use revisions and not tags then you only need to decide if you think the initial clone performance makes it worthwhile to set up `fetch-git`.

Reply:
That is right, fetchGit is morally pure when a revision or tag is specified. fetch-git is intended as a performance improvement, plus a minor restriction in that it doesn't allow the impurity of a floating branch fetch.